tests/target-browser/target_browser_test.go

```diff
@@ -0,0 +1,385 @@
+					strconv.Itoa(noOfBackupsToCreate), "true", backupIDValue, "helm_backup_type").Output()
+
+				log.Info("Shell Script Output: ", string(output))
+				backupPlanUIDs = verifyBackupPlansAndBackupsOnNFS(noOfBackupplansToCreate, noOfBackupplansToCreate*noOfBackupsToCreate)
+				time.Sleep(2 * time.Minute) //wait to sync data on target browser
+			})
+		})
+		AfterEach(func() {
+			if isLast {
+				// delete target & remove all files & directories created for this Context - only once After all It in this context
+				deleteTarget()
+				for _, backupPlans := range backupPlanUIDs {
+					_, err := shell.RmRf(targetLocation + "/" + backupPlans)
+					Expect(err).To(BeNil())
+				}
+			}
+		})
+		It("Should filter metadata on BackupPlan and backup", func() {
+			isLast = true
+			backupPlanUID := backupPlanUIDs[0]
+			args := []string{cmdGet, cmdMetadata, flagBackupPlanUIDFlag, backupPlanUID, flagBackupUIDFlag, backupIDValue}
+			args = append(args, commonArgs...)
+			cmd := exec.Command(exeCmd+path.Join(TargetBrowserBinaryLocation, TargetBrowserBinaryName), args...)
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				Fail(fmt.Sprintf("Error to execute command %s", err.Error()))
+			}
+			log.Infof("Metadata is %s", output)
+			re := regexp.MustCompile("(?m)[\r\n]+^.*location.*$")
+			metadata := re.ReplaceAllString(string(output), "")
+			jsonFile, err := os.Open(path.Join(path.Join(currentDir, testDataDirRelPath, "metadata-helm.json")))
+			Expect(err).To(BeNil())
+
+			defer jsonFile.Close()
+
+			expectedMetadata, _ := ioutil.ReadAll(jsonFile)
+			Expect(reflect.DeepEqual(expectedMetadata, metadata))
+		})
+	})
+
+})
+
+func createTarget() {
+	By("Creating target and marking it available")
+	target := fmt.Sprintf("kubectl apply -f %s --namespace %s", path.Join(currentDir, testDataDirRelPath, targetPath), installNs)
+	cmd := exec.Command("bash", "-c", target)
+
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		Fail(fmt.Sprintf("target creation failed %s.", err.Error()))
+	}
+	VerifyTargetStatus(installNs)
+}
+
+func deleteTarget() {
+	target := fmt.Sprintf("kubectl delete -f %s --namespace %s", path.Join(currentDir, testDataDirRelPath, targetPath), installNs)
+	cmd := exec.Command("bash", "-c", target)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		Fail(fmt.Sprintf("target creation failed %s.", err.Error()))
+	}
+}
+
+func runCmdBackupPlan(args []string) []backupPlan {
+	args = append(args, commonArgs...)
+	var output []byte
+	var err error
+	Eventually(func() bool {
+		cmd := exec.Command(exeCmd+path.Join(TargetBrowserBinaryLocation, TargetBrowserBinaryName), args...)
+		log.Info("BackupPlan command is: ", cmd)
+		output, err = cmd.CombinedOutput()
+		if err != nil {
+			log.Errorf(fmt.Sprintf("Error to execute command %s", err.Error()))
+		}
+		log.Infof("BackupPlan data is %s", output)
+		return strings.Contains(string(output), "502 Bad Gateway")
+	}, timeout, interval).Should(BeFalse())
```

Comment:
cmd should not be executed in loop [also retry count is 300.. too much].. check if it's really needed.

Reply:
Add retry upto 5 times